ci/README.md

```diff
@@ -20,21 +20,36 @@ master commit at which the binary was compiled, and `latest` corresponds to a bi
 
 # Building and running tests as a developer
 
-An example basic invocation to build the Envoy static binary (using the Bazel `fastbuild` type) is:
+An example basic invocation to build a developer version of the Envoy static binary (using the Bazel `fastbuild` type) is:
 
 ```bash
-./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.fastbuild'
+./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev.server_only'
 ```
 
-The output artifacts can be found in `/build/envoy/bazel-bin/source/exe/envoy-static` in the
-container.
+The Envoy binary can be found in `/tmp/envoy-docker-build/envoy/source/exe/envoy-fastbuild` on the Docker host. You
+can control this by setting `ENVOY_DOCKER_BUILD_DIR` in the environment, e.g. to
+generate the binary in `~/build/envoy/source/exe/envoy-fastbuild` you can run:
+
+
+```bash
+ENVOY_DOCKER_BUILD_DIR=~/build ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev.server_only'
+```
+
+For a release Envoy binary you can run:
+
+```bash
+./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.release.server_only'
+```
+
+The artifacts can be found in `/tmp/envoy-docker-build/envoy/source/exe/envoy` (or wherever
+`$ENVOY_DOCKER_BUILD_DIR` points).
 
-The `./ci/run_envoy_docker.sh './ci/do_ci.sh <TARGET>'` targets are:
 
 * `bazel.asan` &mdash; build and run tests under `-c dbg --config=asan`.
-* `bazel.fastbuild` &mdash; build Envoy static binary under `-c fastbuild`.
-* `bazel.fastbuild.test` &mdash; build and run tests under `-c fastbuild`.
+* `bazel.dev.server_only` &mdash; build Envoy static binary under `-c fastbuild`.
+* `bazel.dev.test` &mdash; build and run tests under `-c fastbuild`.
 * `bazel.release` &mdash; build Envoy static binary and run tests under `-c opt`.
+* `bazel.release.server_only` &mdash; build Envoy static binary under `-c opt`.
```

Comment:
(discussed this offline, @htuch is going to add binary build to `bazel.dev.test` and rename it to  `bazel.dev`)